app/models/manageiq/providers/openshift/container_manager/orchestration_stack.rb

```diff
@@ -0,0 +1,56 @@
+class ManageIQ::Providers::Openshift::ContainerManager::OrchestrationStack < ManageIQ::Providers::ContainerManager::OrchestrationStack
+  require_nested :Status
+
+  belongs_to :ext_management_system, :foreign_key => :ems_id, :class_name => "ManageIQ::Providers::ContainerManager"
+  belongs_to :container_template, :foreign_key => :orchestration_template_id, :class_name => "ContainerTemplate"
+
+  def self.create_stack(container_template, params, project_name)
+    new(:name                  => container_template.name,
+        :ext_management_system => container_template.ext_management_system,
+        :container_template    => container_template).tap do |stack|
+      stack.send(:add_provider_objects, raw_create_stack(container_template, params, project_name))
+    end
+  end
+
+  def self.raw_create_stack(container_template, params, project_name)
+    container_template.instantiate(params, project_name)
+  rescue => err
+    _log.error "Failed to provision from container template(#{container_template.name}), error: #{err}"
+    raise MiqException::MiqOrchestrationProvisionError, err.to_s, err.backtrace
+  end
+
+  def self.status_class
+    "#{name}::Status".constantize
+  end
+
+  def raw_status
+    done = resources.all? do |obj|
+      miq_class = obj.resource_category
+      find = miq_class.constantize.find_by(:ems_ref => obj.ems_ref) if miq_class
+      obj.update_attributes(:resource_status => 'succeeded') if find
+      miq_class.nil? || find
+    end
+
+    update_attributes(:status => 'succeeded') if done
+    message = done ? "completed" : "in progress"
+
+    self.class.status_class.new(message, nil)
+  end
+
+  def add_provider_objects(objects)
+    self.resources = objects.collect { |object| add_provider_object(object) }
+  end
+  private :add_provider_objects
+
+  def add_provider_object(object)
+    OrchestrationStackResource.new(
```

Comment:
Are these persisted at some point? I can only see the update attributes (`create`?)